### DISCUSSION TITLE
Change current directory

### DISCUSSION THREAD

**Discussion by nick5454** [cake-build/cake#3288] [Q&A] [answered]
I know this is a dumb question, but I need to change the current directory. "cd {path/to.directory}"

How do I change the directory, is that what the WorkingDirectory refers too?

**Comment by augustoproiete** [comment, marked as answer]
Cake is just C#, so you can use [`System.IO.Directory.SetCurrentDirectory`](https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.setcurrentdirectory?view=net-5.0):

```cakescript
Information("Current working dir: {0}", System.IO.Directory.GetCurrentDirectory()); // original

System.IO.Directory.SetCurrentDirectory(@"C:\SomeOtherDirectory\");
Information("Current working dir: {0}", System.IO.Directory.GetCurrentDirectory()); // C:\SomeOtherDirectory\
```

Alternatively, you can set the [`WorkingDirectory`](https://cakebuild.net/api/Cake.Core/ICakeEnvironment/1D024380) property of the [Environment](https://cakebuild.net/api/Cake.Core/ICakeEnvironment/) in the [Context](https://cakebuild.net/api/Cake.Core/CakeContext/):

```cakescript
Task("Build")
    .Does((context) =>
{
    Information("Current working dir: {0}", context.Environment.WorkingDirectory); // original

    context.Environment.WorkingDirectory = @"C:\SomeOtherDirectory\";
    Information("Current working dir: {0}", context.Environment.WorkingDirectory); // C:\SomeOtherDirectory\
});

RunTarget("Build");
```

---

If you want to temporarily switch to a directory, run something, then go back to the original working directory there's an addin called [`Cake.DoInDirectory`](https://cakebuild.net/extensions/cake-doindirectory/) that makes this easier:

```cakescript
#addin "nuget:?package=Cake.DoInDirectory&version=4.0.1"

Information("Current working dir: {0}", System.IO.Directory.GetCurrentDirectory()); // original

DoInDirectory(@"C:\SomeOtherDirectory\", () =>
{
    Information("Current working dir: {0}", System.IO.Directory.GetCurrentDirectory()); // C:\SomeOtherDirectory\
});

Information("Current working dir: {0}", System.IO.Directory.GetCurrentDirectory()); // original
```

**Reply by nick5454** [reply]
Thanks so ,much that DoInDirectory is great!

**Reply by augustoproiete** [reply]
Indeed. Credits to @pitermarx :portugal:
